jumpscale/packages/vdc_dashboard/chats/presearch.py

```diff
@@ -0,0 +1,49 @@
+from jumpscale.sals.chatflows.chatflows import StopChatFlow, chatflow_step
+from jumpscale.packages.vdc_dashboard.sals.solutions_chatflow import SolutionsChatflowDeploy
+from jumpscale.packages.vdc_dashboard.sals.vdc_dashboard_sals import get_deployments
+
+
+class PresearchDeploy(SolutionsChatflowDeploy):
+    SOLUTION_TYPE = "presearch"
+    title = "Presearch"
+    HELM_REPO_NAME = "marketplace"
+    steps = ["get_release_name", "create_subdomain", "set_config", "install_chart", "initializing", "success"]
+
+    CHART_LIMITS = {
+        "Silver": {"cpu": "2000m", "memory": "2024Mi"},
+        "Gold": {"cpu": "3000m", "memory": "3072Mi"},
+        "Platinum": {"cpu": "4000m", "memory": "4096Mi"},
+    }
+
+    def _check_uniqueness(self):
+        username = self.user_info()["username"]
+        self.md_show_update("Preparing the chatflow...")
+        if get_deployments(self.SOLUTION_TYPE, username):
+            raise StopChatFlow("You can only have one Presearch solution per VDC")
+
+    def get_release_name(self):
+        self._check_uniqueness()
+        super().get_release_name()
+
+    def _enter_registration_code(self):
+
+        form = self.new_form()
```

Comment:
move to string_ask

Reply:
@xmonader It is doable, not an issue ! But that is my concern again, why do you have double standards ? 

- You lecture people not have SSH on pods whereas your own charts have SSH to pods ? See this chart https://github.com/threefoldtech/vdc-solutions-charts/blob/master/charts/gitea/values.yaml

![image](https://user-images.githubusercontent.com/25789764/108060626-8d0cb380-7070-11eb-9582-804447d9742b.png)

- It is a trivial matter to use a form or string ask. This PR which was approved had 2 questions , we could have used string.ask as well ? See this https://github.com/threefoldtech/js-sdk/commit/a48f8f3cf0dacabccc0188da1795623918ad8ab3

Reply:
You really took about an hour to search for older PRs and non related commits to prove your point on a two-lines change? what a waste of time.

Reply:
True ! Let us avoid that in the future.